Adapter/ImageMediaAdapter.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\SyliusConsumerBundle\Adapter;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Sulu\Bundle\MediaBundle\Entity\Collection;
+use Sulu\Bundle\MediaBundle\Entity\File;
+use Sulu\Bundle\MediaBundle\Entity\FileVersion;
+use Sulu\Bundle\MediaBundle\Entity\FileVersionMeta;
+use Sulu\Bundle\MediaBundle\Entity\Media;
+use Sulu\Bundle\MediaBundle\Entity\MediaInterface;
+use Sulu\Bundle\MediaBundle\Entity\MediaType;
+use Sulu\Bundle\MediaBundle\Media\Storage\StorageInterface;
+use Sulu\Bundle\SyliusConsumerBundle\Exception\ImageDownloadFailedException;
+use Sulu\Bundle\SyliusConsumerBundle\Payload\ImagePayload;
+use Sulu\Bundle\SyliusConsumerBundle\Repository\ImageMediaBridgeRepositoryInterface;
+use Sulu\Component\Media\SystemCollections\SystemCollectionManagerInterface;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class ImageMediaAdapter implements ImageAdapterInterface
+{
+    private const IMAGE_MEDIA_TYPE = 2;
+
+    /**
+     * @var ImageMediaBridgeRepositoryInterface
+     */
+    private $mediaBridgeRepository;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    /**
+     * @var HttpClientInterface
+     */
+    private $httpClient;
+
+    /**
+     * @var StorageInterface
+     */
+    private $storage;
+
+    /**
+     * @var SystemCollectionManagerInterface
+     */
+    private $systemCollectionManager;
+
+    /**
+     * @var string
+     */
+    private $syliusBaseUrl;
+
+    /**
+     * @var string
+     */
+    private $collectionKey;
+
+    public function __construct(
+        ImageMediaBridgeRepositoryInterface $mediaBridgeRepository,
+        EntityManagerInterface $entityManager,
+        HttpClientInterface $client,
+        StorageInterface $storage,
+        SystemCollectionManagerInterface $systemCollectionManager,
+        string $syliusBaseUrl,
+        string $collectionKey
+    ) {
+        $this->mediaBridgeRepository = $mediaBridgeRepository;
+        $this->entityManager = $entityManager;
+        $this->syliusBaseUrl = $syliusBaseUrl;
+        $this->httpClient = $client;
+        $this->storage = $storage;
+        $this->systemCollectionManager = $systemCollectionManager;
+        $this->collectionKey = $collectionKey;
+    }
+
+    public function synchronize(ImagePayload $payload): void
+    {
+        $this->handlePayload($payload);
+
+        // Needed to use medias in other adapters
+        $this->entityManager->flush();
+    }
+
+    public function remove(int $id): void
+    {
+        $this->mediaBridgeRepository->removeById($id);
```

Comment:
Should the real image (Media entity) here be removed too?

Reply:
hmm hard to say - but would say yes